Resteasy Reactive first POST with InputStream hang

**Describe the bug**
Using resteasy-reactive the first POST with an input stream hang forever

**Expected behavior**
Return the payload

**Actual behavior**
The socket stay connected forever if the client doesn't implement a timeout

**To Reproduce**
Clone this repo:

[https://github.com/masini/test-case-resteasy-reactive-first-post-bug](https://github.com/masini/test-case-resteasy-reactive-first-post-bug)

Steps to reproduce the behavior:
1. mvn quarkus:dev
2. attach e debugger
3. run the curl in the readme.md

**Configuration**
```properties
####HTTP #########
quarkus.http.port=8082
quarkus.http.ssl-port=9081

quarkus.http.access-log.enabled=true
quarkus.http.root-path=/resources

####SECURITY Start#########
quarkus.http.auth.basic=true
quarkus.security.users.embedded.enabled=true
quarkus.security.users.embedded.plain-text=true
quarkus.security.users.embedded.users.admin=admin
quarkus.security.users.embedded.roles.admin=TERMINAL_ROLE,APP_ROLE,TECHNICAL_ROLE

quarkus.http.ssl.certificate.key-store-file=tls/keystore.p12
quarkus.http.ssl.certificate.key-store-password=infogroup
quarkus.http.ssl.certificate.key-store-file-type=PKCS12
```

**Environment (please complete the following information):**
 - `uname -a`: Darwin WMICTLM1P.lan 19.6.0 Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64 
 - `java -version`: 
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.10+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.10+9, mixed mode)
 - Quarkus version or git rev: 1.12.0.Final (also tried with 1.12.1.Final, pushed this version)
 - `mvnw --version`: 
Maven home: /Users/ictlm1/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.10, vendor: AdoptOpenJDK, runtime: /Applications/sviluppo/java/jdk-11.0.10+9/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.7", arch: "x86_64", family: "mac"

**Additional context**
This is a reproducer of a real problem in a real application.

In the real application this happens also without debugger and with the quarkus-run.jar, the reproducer instead is able to reproduce only with a debugger attached and not always. I think it's a matter of timings, but I'm not sure, I need to debug more.

Hi @geoand, introducing the bom and SNAPSHOT version:

```xml
    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
    <quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>

```
I got this from Maven:
```java
[ERROR]     'dependencies.dependency.version' for io.quarkus:quarkus-resteasy-reactive-jackson:jar is missing. @ line 37, column 17
```
if I add the version 999-SNAPSHOT to that extension and then I launch quarkus:dev I got this:

```java
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: io.quarkus:quarkus-resteasy-reactive:jar:, io.quarkus:quarkus-jackson:jar:, io.quarkus:quarkus-jsonp:jar:: Could not find artifact io.quarkus:quarkus-resteasy-reactive:jar: in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:424)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:340)
	... 7 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.quarkus:quarkus-resteasy-reactive:jar: in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:48)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:369)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

```
finally using Maven offline I got back to the starting point, Quarkus start and exit:

```java
2021-03-10 07:16:10,280 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2021-03-10 07:16:10,698 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 418ms
2021-03-10 07:16:10,752 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 0.624s. 
2021-03-10 07:16:10,752 INFO  [io.quarkus] (main) Installed features: [cdi]
2021-03-10 07:16:10,755 INFO  [io.quarkus] (Quarkus Shutdown Thread) Quarkus stopped in 0.001s

```
Any idea ? Too big to be true.

--- a/independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/VertxResteasyReactiveRequestContext.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/VertxResteasyReactiveRequestContext.java
@@ -217,6 +217,7 @@ public class VertxResteasyReactiveRequestContext extends ResteasyReactiveRequest
 
     @Override
     public ServerHttpResponse setReadListener(ReadCallback callback) {
+        request.pause();
         if (continueState == ContinueState.REQUIRED) {
             continueState = ContinueState.SENT;
             response.writeContinue();
@@ -233,6 +234,7 @@ public class VertxResteasyReactiveRequestContext extends ResteasyReactiveRequest
                 callback.done();
             }
         });
+        request.resume();
         return this;
     }
 

